src/Teams/Models/IRepository.cs

```diff
@@ -0,0 +1,17 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Teams.Models
+{
+    interface IRepository<T> where T : class
+    {
+        IEnumerable<T> GetAll();
+        T GetById(object id);
+        void Insert(T obj);
+        void Update(T obj);
+        void Delete(object id);
+        void Save();
```

Comment:
All operation to the DB should be Task returning instead of void and naming convention should be according to {MethodName}Async. Example: Task SaveAsync()